PlaywrigthSpecFlow/Features/Account/AccountSteps.cs

```diff
@@ -0,0 +1,47 @@
+using PlaywrigthSpecFlow.API.Features.Account;
+using TechTalk.SpecFlow;
+
+namespace PlaywrigthSpecFlow.Features.Account
+{
+    [Binding]
+    public class AccountSteps
+    {
+        private readonly FeatureContext _featureContext;
+
+        public AccountSteps(FeatureContext featureContext)
+        {
+            _featureContext = featureContext;
+        }
+
+        [BeforeFeature(@"ICreateAccountByAPI")]
+        public static async Task WhenICreateAccountByAPI(FeatureContext featureContext)
+        {
+            AccountPresetup accountPresetup = new AccountPresetup();
+            await accountPresetup.AccountApiPresetup();
+            featureContext["AccountApiPresetup"] = accountPresetup;
+        }
+
+        [Then(@"I get success status code from API")]
+        public void ThenISeeSuccessStatusCode()
+        {
+            var presetup = _featureContext.Get<AccountPresetup>("AccountApiPresetup");
+            Assert.That(presetup.UserId, Is.Not.Null, "Account not created");
+        }
+
+        //TODO: add cleanup
+        [AfterFeature(@"IDeleteAccountByAPI")]
+        public static async Task WhenICleanupAccountByAPI(FeatureContext featureContext)
+        {
+            var presetup = featureContext.Get<AccountPresetup>("AccountApiPresetup");
+            await presetup.AccountApiCleanup();
+        }
+
+        [Then(@"I get message that account deleted after cleanup")]
+        public async Task ThenIGetMessageThatAccountDeletedAsync()
+        {
+            var presetup = _featureContext.Get<AccountPresetup>("AccountApiPresetup");
+            await presetup.AccountApiCleanup();
+            Assert.That(presetup.UserId, Is.Not.Null, "Account is not deleted");
```

Comment:
This step         [Then(@"I get message that account deleted after cleanup")] is called BEFORE  [AfterFeature(@"IDeleteAccountByAPI")]
So what you do - removing account twice in both these steps.
This assert  Assert.That(presetup.UserId, Is.Not.Null, "Account is not deleted"); is wrong as it verifies that UserId Is.Not.Null and if it fails (which means UserId IS Null) then it shows description  "Account is not deleted", so if UserId IS Null then error  "Account is not deleted" and if UserId IS NOT Null then this assert passes.
Why it passes after you remove account - because value UserId is saved in AccountPresetup when user is created and this value is never deleted from UserId variable even if you deleted user via API

Reply:
thank you, i decided to add this additional step to the test's steps to be aligned with the logic,
because looking into this test i don't see that after creation User id it is deleted automatically. yes, we created method for deletion but i don't see where it calls in the test. 

maybe i am looking into the wrong place sorry or cannot understand smth.
![image](https://github.com/user-attachments/assets/adc879cc-4f42-4933-82a2-b6831a45216e)

Reply:
maybe yes, we have set up attribute 'After feature' and User is deleted automatically after creation but i wanted to add this verification step to the test, but it is not clear for now how i can do this 
![image](https://github.com/user-attachments/assets/de1f2967-ac72-42fe-9369-8df9f07f2aba)

Reply:
put breakpoints into the code and run in debug
![image](https://github.com/user-attachments/assets/96d15d50-9fab-4e98-aed3-84cece834048)
![image](https://github.com/user-attachments/assets/f72e40da-0859-4781-953d-eb592c352d98)
here as well 
![image](https://github.com/user-attachments/assets/94b385b6-f7b4-42b8-bff6-3dacd3fd2f86)

and then use Continue and Step Over to navigate and you will see that it executes like:
[BeforeFeature(@"ICreateAccountByAPI")]
[Then(@"I get success status code from API")]
[AfterFeature(@"ICreateAccountByAPI")]

if account is not removed, test will be green, as removal happens in AfterFeature which is not related to the test itself, it is Feature level, so you can see results in Debug only (in this particular configuration of the test)

Reply:
thank you, i see my problem, it is very useful when you get feedback form the mentor )

Reply:
if you want to verify account removal, you can change test so Before/After Feature are empty and there are 3 steps like
Given I create account via API
When I remove account
Then Account is removed
and use endpoints
1) create
2) generate token, get user info (optional), remove
3) get user info and verify response "User does not exist)
So it is separate test case

Given example is not complete. Ideally it should be something like
BeforeFeature - create
Several test cases with Given/When/Then - use account's token in UI test
AfterFeature - remove